ambari-common/src/main/python/resource_management/libraries/functions/mpack_manager_helper.py

```diff
@@ -0,0 +1,91 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Ambari Agent
+
+"""
+
+import os
+from instance_manager import *
+
+CONFIG_DIR_KEY_NAME = 'config_dir'
+PATH_KEY_NAME = 'path'
+COMPONENTS_PLURAL_KEY_NAME = 'components'
+COMPONENT_INSTANCES_PLURAL_KEY_NAME = 'component-instances'
+
+
+def get_component_conf_path(mpack_name, instance_name, module_name, components_instance_type,
+                            subgroup_name='default', component_instance_name='default'):
+  """
+  :returns the single string that contains the path to the configuration folder of given component instance
+  :raises ValueError if the parameters doesn't match the mpack or instances structure
+  """
+
+  conf_json = get_conf_dir(mpack_name, instance_name, subgroup_name, module_name,
+                           {components_instance_type: [component_instance_name]})
+
+  return conf_json[COMPONENTS_PLURAL_KEY_NAME][components_instance_type.lower()][COMPONENT_INSTANCES_PLURAL_KEY_NAME][
+    component_instance_name][CONFIG_DIR_KEY_NAME]
+
+
+def get_component_target_path(mpack_name, instance_name, module_name, components_instance_type,
+                              subgroup_name='default', component_instance_name='default'):
+  """
+  :returns the single string that contains the path to the mpack component folder of given component instance
+  :raises ValueError if the parameters doesn't match the mpack or instances structure
+  """
+
+  instances_json = list_instances(mpack_name, instance_name, subgroup_name, module_name,
+                                  {components_instance_type: [component_instance_name]})
+
+  return instances_json[COMPONENTS_PLURAL_KEY_NAME][components_instance_type.lower()][
+    COMPONENT_INSTANCES_PLURAL_KEY_NAME][component_instance_name][PATH_KEY_NAME]
+
+
+def get_component_home_path(mpack_name, instance_name, module_name, components_instance_type,
+                            subgroup_name='default', component_instance_name='default'):
+  """
+  :returns the single string that contains the path to the module component folder of given component instance
+  :raises ValueError if the parameters doesn't match the mpack or instances structure
+  """
+
+  component_path = get_component_target_path(mpack_name=mpack_name, instance_name=instance_name,
+                                             subgroup_name=subgroup_name,
+                                             module_name=module_name, components_instance_type=components_instance_type,
+                                             component_instance_name=component_instance_name)
+
+  return os.readlink(component_path)
+
+
+def create_component_instance(mpack_name, mpack_version, instance_name, module_name, components_instance_type,
```

Comment:
Nit-picking rename components_instance_type to component_name

Reply:
I think the "component_type" is better then get_component_name since we also have component_instance_name.